knowledge-graph/src/main/scala/ch/datascience/knowledgegraph/datasets/rest/BaseDetailsFinder.scala

```diff
@@ -103,6 +104,59 @@ private class BaseDetailsFinder(
         |""".stripMargin
   )
 
+  def findKeywords(identifier: Identifier): IO[List[Keyword]] =
+    queryExpecting[List[Keyword]](using = queryKeywords(identifier)).flatMap(s => ME.pure(s))
+
+  private def queryKeywords(identifier: Identifier) = SparqlQuery(
+    name = "ds by id - keyword details",
+    Set(
+      "PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>",
+      "PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>",
+      "PREFIX schema: <http://schema.org/>"
+    ),
+    s"""|SELECT DISTINCT ?datasetId ?keyword
+        |WHERE {
+        |  {
+        |    SELECT ?topmostSameAs
```

Comment:
Why do we need this whole section with the topmostSameAs?

Reply:
Oh yes for some reason I thought that this would gather all the keywords from all the versions of the dataset, but the identifier is enough to filter